packages/devtools_app/lib/src/memory/flutter/memory_filter.dart

```diff
@@ -0,0 +1,391 @@
+                              value: controller.filterZeroInstances,
+                              onChanged: (value) {
+                                setState(() {
+                                  controller.filterZeroInstances = value;
+                                });
+                              }),
+                          const Text('Hide Classes with No Instances'),
+                        ],
+                      ),
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.start,
+                        children: [
+                          Checkbox(
+                              value: controller.filterLibraryNoInstances,
+                              onChanged: (value) {
+                                setState(() {
+                                  controller.filterLibraryNoInstances = value;
+                                });
+                              }),
+                          const Text('Hide Library with No Instances'),
+                        ],
+                      ),
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.start,
+                        children: [
+                          const Padding(padding: EdgeInsets.only(top: 30)),
+                          Text('Hide Libraries or Packages '
+                              '(${controller.filteredLibrariesByGroupName.length}):'),
+                        ],
+                      ),
+                      SizedBox(
+                        height: constraints.maxHeight / 4,
+                        child: ListView(
+                          controller: _letters,
+                          children: controller.filteredLibrariesByGroupName.keys
+                              .map((String key) {
+                            return CheckboxListTile(
+                              title: Text(key),
+                              dense: true,
+                              value: controller
+                                  .filteredLibrariesByGroupName[key].first.hide,
+                              onChanged: (bool value) {
+                                setState(() {
+                                  for (var filter in controller
+                                      .filteredLibrariesByGroupName[key]) {
+                                    filter.hide = value;
+                                  }
+                                });
+                              },
+                            );
+                          }).toList(),
+                        ),
+                      ),
+                      const Padding(padding: EdgeInsets.only(top: 40)),
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                        children: [
+                          DialogOkButton(
+                            () {
+                              // Re-generate librariesFiltered
```

Comment:
move this to a separate helper method as this is a lot of business logic to include inline in the build method.